src/model/src/SolidShapes.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2016 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Silvio Traversaro
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include <iDynTree/Model/SolidShapes.h>
+#include <iDynTree/Model/Model.h>
+
+namespace iDynTree
+{
+    SolidShape::~SolidShape()
+    {
+    }
+
+    bool SolidShape::isSphere() const
+    {
+        return (dynamic_cast<const Sphere*>(this) != 0);
+    }
+
+    Sphere* SolidShape::asSphere()
+    {
+        return dynamic_cast<Sphere*>(this);
+    }
+
+    const Sphere* SolidShape::asSphere() const
+    {
+        return dynamic_cast<const Sphere*>(this);
+    }
+
+    bool SolidShape::isCylinder() const
+    {
+        return (dynamic_cast<const Cylinder*>(this) != 0);
+    }
+
+    Cylinder * SolidShape::asCylinder()
+    {
+        return dynamic_cast<Cylinder*>(this);
+    }
+
+    const Cylinder * SolidShape::asCylinder() const
+    {
+        return dynamic_cast<const Cylinder*>(this);
+    }
+
+    bool SolidShape::isBox() const
+    {
+        return (dynamic_cast<const Box*>(this) != 0);
+    }
+
+    Box * SolidShape::asBox()
+    {
+        return dynamic_cast<Box*>(this);
+    }
+
+    const Box * SolidShape::asBox() const
+    {
+        return dynamic_cast<const Box*>(this);
+    }
+
+    bool SolidShape::isExternalMesh() const
+    {
+        return (dynamic_cast<const ExternalMesh*>(this) != 0);
+    }
+
+    ExternalMesh * SolidShape::asExternalMesh()
+    {
+        return dynamic_cast<ExternalMesh*>(this);
+    }
+
+    const ExternalMesh * SolidShape::asExternalMesh() const
+    {
+        return dynamic_cast<const ExternalMesh*>(this);
+    }
+
+
+    Sphere::~Sphere()
+    {
+    }
+
+    SolidShape* Sphere::clone()
+    {
+        return static_cast<SolidShape*>(new Sphere(*this));
```

Comment:
I removed and no one complained. We can reinsert it when we enable the warnings.